hazelcast/src/main/java/com/hazelcast/map/impl/recordstore/DefaultRecordStore.java

```diff
@@ -1281,13 +1281,30 @@ public void removeRecord0(Data key, @Nonnull Record record, boolean backup) {
         storage.removeRecord(key, record);
     }
 
+    public Record removeByKey(Data key, boolean backup) {
+        Record record = getRecord(key);
+        mutationObserver.onRemoveRecord(key, record, backup);
+        removeKeyFromExpirySystem(key);
+        storage.removeRecord(key, record);
+        return record;
+    }
```

Comment:
> 
```suggestion
    public Record removeByKey(Data key, boolean backup) {
        Record record = getRecord(key);
        removeRecord0(key, record, backup);
        return record;
    }
```

Reply:
Addressed in https://github.com/hazelcast/hazelcast/pull/26065/commits/1f85d1fd9de2c3f6d6af02faeafc9ecfe7fac6f5